dev-python/m2crypto/m2crypto-0.31.0-r2.ebuild

```diff
@@ -43,12 +43,17 @@ python_compile() {
 	# setup.py looks at platform.machine() to determine swig options.
 	# For exotic ABIs, we need to give swig a hint.
 	# https://bugs.gentoo.org/617946
-	# TODO: Fix cross-compiles
 	local -x SWIG_FEATURES=
 	case ${ABI} in
 		x32) SWIG_FEATURES="-D__ILP32__" ;;
 	esac
-	distutils-r1_python_compile --openssl="${EPREFIX}"/usr
+	distutils-r1_python_compile --openssl="${EROOT}"/usr
+}
+
+src_prepare() {
+	# Switch out the compiler and it's arguments (hence the second sed) to the current build compiler instead of just the `cpp`
+	sed -i -e "s:^        pid = subprocess.Popen(\['cpp', '-Wp,-v', '-'\],:        pid = subprocess.Popen(\['`tc-getCPP | sed "s: :', ':"`', '-Wp,-v', '-'\],:" setup.py
```

Comment:
No way am I merging this ugly sed. Maybe patch setup.py to grab the cpp binary from the CPP environment variable instead.

Reply:
Yes, it is ugly, I agree. I am able to change this, but the CPP environment variable is not enough, because I also need to include the command arguments that the `tc-getCPP` outputs.
I can create other environment variables to make this line more clear.